File: BackEnd/proyecto/src/main/java/com/example/proyecto/model/Alumno.java
package com.example.proyecto.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.util.LinkedHashSet;
import java.util.Set;

@Getter
@Setter
@ToString
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "alumno")
public class Alumno {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Integer id;

    @Column(name = "nombre")
    private String nombre;

    @Column(name = "apellidos")
    private String apellidos;

    @Column(name = "foto")
    private String foto;

//    @ManyToOne(fetch = FetchType.LAZY, optional = false)
//    @JoinColumn(name = "id_curso", nullable = false)
//    private Curso idCurso;

    @Column(name = "id_curso")
    private Integer idCurso;

    @Column(name = "dni")
    private String dni;

    @OneToMany(mappedBy = "idAlumno")
    @JsonIgnore
    private Set<Falta> faltas = new LinkedHashSet<>();

//    @OneToMany(mappedBy = "alumno", cascade = CascadeType.ALL, orphanRemoval = true)
//    @JsonIgnore
//    private Set<TutorLegalAlumno> tutoresLegales = new LinkedHashSet<>();

    @ManyToMany(mappedBy = "alumnos")
    @JsonIgnore
    private Set<TutorLegal> tutorLegales= new LinkedHashSet<>();

}

File: FrontEnd/app/src/main/java/com/example/tycep_fe/models/Alumno.kt
package com.example.tycep_fe.models

import com.google.gson.annotations.SerializedName

data class Alumno(
    @SerializedName("id")
    val id:Int,
    @SerializedName("nombre")
    val nombre:String,
    @SerializedName("apellidos")
    val apellidos:String,
    @SerializedName("foto")
    val foto:String,
    @SerializedName("idCurso")
    val idCurso:Int,
    @SerializedName("dni")
    val dni:String,
    @SerializedName("faltas")
    var faltas:Set<Falta>?
)
